python/cudf/cudf/core/udf/groupby_utils.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+
+import math
+import os
+
+import cupy as cp
+import numpy as np
+from numba import cuda, types
+from numba.cuda.cudadrv.devices import get_context
+from numba.np import numpy_support
+from numba.types import Record
+
+import cudf.core.udf.utils
+from cudf.core.udf.groupby_typing import (
+    SUPPORTED_GROUPBY_NUMPY_TYPES,
+    Group,
+    GroupType,
+)
+from cudf.core.udf.templates import (
+    group_initializer_template,
+    groupby_apply_kernel_template,
+)
+from cudf.core.udf.utils import (
+    _get_extensionty_size,
+    _get_kernel,
+    _get_ptx_file,
+    _get_udf_return_type,
+    _supported_cols_from_frame,
+    _supported_dtypes_from_frame,
+)
+from cudf.utils.utils import _cudf_nvtx_annotate
+
+dev_func_ptx = _get_ptx_file(os.path.dirname(__file__), "function_")
```

Comment:
Rather than duplicating this logic in `cudf/__init__.py`, could we import this variable there? I believe we use the path there as well to set up the numba env.

Reply:
This has been fixed and `_setup_numba_linker` does not redetermine the PTX file path anymore instead expecting the caller to do so.